src/components/Checkout/KivaCreditPayment.vue

```diff
@@ -47,29 +48,56 @@ export default {
 					}
 				}).catch(errorResponse => {
 					this.$emit('updating-totals', false);
-					console.error(errorResponse);
+					this.showCheckoutError(errorResponse);
 				});
 		},
 		checkoutCreditBasket() {
 			this.checkoutBasket(false, this.useAsyncCheckout)
-				.then(transactionResult => {
-					if (typeof transactionResult !== 'object') {
-						// succesful validation
-						this.$kvTrackEvent('basket', 'Kiva Checkout', 'Success', transactionResult);
-						// Complete transaction handles additional analytics + redirect
-						this.$emit('complete-transaction', transactionResult);
+				.then(async transactionResult => {
+					if (this.useAsyncCheckout && typeof transactionResult !== 'object') {
+						await pollForFinishedCheckout({
+							apollo: this.apollo,
+							transactionSagaId: transactionResult,
+						})
+							.then(checkoutStatusResponse => {
+								// eslint-disable-next-line max-len
+								this.handleSuccessfulCheckout(checkoutStatusResponse?.data?.checkoutStatus?.receipt?.checkoutId);
```

Comment:
I see that `checkoutId` has optional chaining, does there need to be a fallback other than `undefined`?